### ISSUE TITLE
fixture id cannot be the same as dom-module-id (Polymer 2.x)

### ISSUE THREAD

**Issue by pghalliday** [Polymer/web-component-tester#503]
I've been porting a project to the Polymer 2.x release candidate and came across this hard to trace bug (at least I think it's a bug).

In my tests I was often creating fixtures with id's that were the same as the id's of the dom-modules under test. It worked fine  on every browser I tested it on. However when I switched to Polymer 2.x with the following versions:

Polymer CLI: 0.18.0-pre.13
web-component-tester: v6.0.0-prerelease.6
webcomponentsjs: webcomponents/webcomponentsjs#1.0.0-rc.5

Everything ran fine on chrome, but every other browser failed with the following error (in this case from safari):

```
Error: document.getElementById(fixtureId).create is not a function. (In 'document.getElementById(fixtureId).create(model)', 'document.getElementById(fixtureId).create' is undefined)
  <unknown> at auth-pending.html:22
```

It took a long time but eventually I figured out that the `document.getElementById(fixtureId)` was returning the `dom-module` element and not the fixture. Changing the fixture id to something different to the `dom-module` id fixed the problem.

For now I'm going to change all my fixture id's to camelCase as that will work around the problem (all `dom-module` id's must contain a dash I think). But this is a difference in behavior from the Polymer 1.x set up that was somewhat surprising and hard to debug.

**Comment by justinfagnani** [comment]
If the dom-module and text-fixture instances with the same id are in the same scope (same document, or same shadow root), then yes, they should cause collisions, and getElementById() will return the first in document order.

This is pretty much working as intended, but I am curious why this worked with Polymer 1.

My guess here is that it's the new HTML imports polyfill, which places all the imported DOM into the main document, rather than `Document` instances, in order to work with Safari's native Custom Elements implementation. If I'm correct, you probably don't see this issue on Chrome.

I'll leave the issue open, at least so we can verify and open a documentation bug noting to be very careful to avoid id collisions outside of templates/shadow roots.

Can you confirm that this error doesn't happen on Chrome?

@arthurevans and @katejeffreys for heads up on docs.

**Comment by pghalliday** [comment]
You're correct, it doesn't happen on chrome. It did occur to me that the fixture method could limit its search to test-fixture elements. But otherwise yes, getElementById should return the first matching element.
